backend/composer/models.py

```diff
@@ -120,6 +123,24 @@ def get_queryset(self):
         )
 
 
+class AnnotatedConnectionLayerQuerySet(models.QuerySet):
+    """
+    Useful for filtering - GenericConnectionStatementFilter
+    """
+    def annotate_with_ontology_uri(self):
```

Comment:
remark: @D-GopalKrishna it is nice to use `annotate` to create a new result (column) in the query

critical: @D-GopalKrishna however I also see you using the annotated result in a filter. This is not a good idea because it breaks the indexing.... unless you create a functional index in the postgresql database.. As expllained last week please construct a union for this: query filter by `anatomical_entities__simple_entity__ontology_uri` union filter by `anatomical_entities__region_layer__region__ontology_uri` union filter by `anatomical_entities__region_layer__layer__ontology_uri`